### DISCUSSION TITLE
How to use Ollama in the siderbar?

### DISCUSSION THREAD

**Discussion by NextStep-IM** [end-4/dots-hyprland#535] [Q&A]
Sorry if I missed any instruction or/and if this is a stupid question. I ran `ollama run llama3` in the terminal and the model ran successfully, but I don't know what to do with the sidebar.

**Comment by end-4** [comment]
@Soliprem

**Comment by Soliprem** [comment]
Under GPT, you should simply be able to select it from the drop-down menu, like any other compatible model. If you wish for it to be the permanent default, the easiest way (and what I do) is placing it first in the services/gpt.js file and renaming it to openai (you'll probably have to comment out the actual openai service though. I do, because I personally don't use it anyways)

**Reply by end-4** [reply]
that's a more simple answer than i thought
also i should add a config option for default gpt provider and sidebar page order

**Reply by Soliprem** [reply]
That's a feature I agree would be very nice. I'm not entirely sure of how you'd implement it in terms of UI, though. One interesting idea could be to make changes the order persistent. So the last selected option becomes the default one, even across reloads

**Reply by end-4** [reply]
oh actually, `ai.defaultGPTProvider` in user options is already there

**Reply by Soliprem** [reply]
since it's already there, it'd be nice to find a way to implement it in the UI. For example, by changing the UI. I think the idea of making the selection persistent is easiest. Could probably do so with a sed command in the ProviderSwitcher function, in the modules/sideleft/apis/chatgpt.js file. The sed or awk would be simply tasked to replace the value of _currentProvider to the current id
